### PR TITLE
Adds an additional scheme for staging

### PR DESCRIPTION
**Pull request by alvinyongho**
Define a new build sheme in order to stage changes against a separate AWS project.
This allows developers to stage schematic changes and test changes to an AWS project safely (before applying those changes to the prod project) so that we can avoid breaking the production AWS project in future releases

### CODE REVIEW

File: Motcha/AppConfiguration.swift [hunk 1]
```diff
@@ -8,9 +8,24 @@
 
 import Foundation
 
-struct AppConfiguration {
-    static let region = "us-east-1"
-    static let appInstanceArn = "arn:aws:chime:us-east-1:431557582387:app-instance/17145b22-ded4-4762-8e64-344d26c323aa"
-    static let apiGatewayInvokeUrl = "https://3qe9qan1ai.execute-api.us-east-1.amazonaws.com/prod/"
-    static let pinpointAppArn = "PINPOINT_APPLICATION_ARN"
+struct AwsConfig {
+    let region: String
```
Comment (jwarmkessel) [inline Motcha/AppConfiguration.swift:12]: Can you add comments for each property?

File: Motcha/AppConfiguration.swift [hunk 2]
```diff
@@ -8,9 +8,24 @@
...
+    let appInstanceArn: String
+    let apiGatewayInvokeUrl: String
+    let pinpointAppArn: String
 }
+
+let DEV_CONFIG = AwsConfig(
```
Comment (jwarmkessel) [inline Motcha/AppConfiguration.swift:18]: Should we name as "SANDBOX_AWS_CONFIG" or "SANDBOX_CONFIG"

File: Motcha/AppConfiguration.swift [hunk 3]
```diff
@@ -8,9 +8,24 @@
...
+    pinpointAppArn: "PINPOINT_APPLICATION_ARN")
+
+let PROD_CONFIG = AwsConfig(
+    region: "us-east-1",
+    appInstanceArn: "arn:aws:chime:us-east-1:431557582387:app-instance/17145b22-ded4-4762-8e64-344d26c323aa",
+    apiGatewayInvokeUrl: "https://3qe9qan1ai.execute-api.us-east-1.amazonaws.com/prod/",
+    pinpointAppArn: "PINPOINT_APPLICATION_ARN")
+
+let config = Bundle.main.object(forInfoDictionaryKey: "Config") as! String
+let AppConfiguration = config == "Staging" ? DEV_CONFIG : PROD_CONFIG
```
Comment (jwarmkessel) [inline Motcha/AppConfiguration.swift:31]: Should "Staging" be lower case and should/could it be "sandbox" Can we also use an private 

```private enum Constants { static let staging = "staging" }```

File: Motcha/AppDelegate.swift [hunk 4]
```diff
@@ -19,7 +23,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         do {
             try Amplify.add(plugin: AWSCognitoAuthPlugin())
             try Amplify.add(plugin: AWSS3StoragePlugin())
-            try Amplify.configure()
+            
+            let config = Bundle.main.object(forInfoDictionaryKey: "Config") as! String            
+            // Default production amplify configuration.
+            guard let configurationFile = config == "Staging" ?
```
Comment (jwarmkessel) [inline Motcha/AppDelegate.swift:29]: So we are now defaulting to the staging environment?

Comment (jwarmkessel) [reply Motcha/AppDelegate.swift:29]: We probably want to create a project variable that is managed by the scheme so when we create builds we will automatically set the correct configuration.

File: Motcha/AppDelegate.swift [hunk 5]
```diff
@@ -19,7 +23,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         do {
             try Amplify.add(plugin: AWSCognitoAuthPlugin())
             try Amplify.add(plugin: AWSS3StoragePlugin())
-            try Amplify.configure()
+            
+            let config = Bundle.main.object(forInfoDictionaryKey: "Config") as! String
```
Comment (jwarmkessel) [inline Motcha/AppDelegate.swift:27]: Can we separate this code into its own function?

File: Motcha/Info.plist [hunk 6]
```diff
@@ -22,8 +18,14 @@
 	<string>1.0</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
+	<key>Config</key>
```
Comment (jwarmkessel) [inline Motcha/Info.plist:21]: Can this be named to something more explicit?

File: Motcha/configs/dev-amplifyconfiguration.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (jwarmkessel) [inline Motcha/configs/dev-amplifyconfiguration.json:1]: Can this file be renamed to "sandbox-..."

File: Motcha.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -664,14 +759,14 @@
...
-				DEVELOPMENT_TEAM = BMBFCQ6T88;
+				DEVELOPMENT_TEAM = JH979G75UV;
 				INFOPLIST_FILE = Motcha/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 16.0;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.mochamatcha.motcha;
+				PRODUCT_BUNDLE_IDENTIFIER = app.motcha;
```
Comment (jwarmkessel) [inline Motcha.xcodeproj/project.pbxproj:769]: This bundle identifier doesn't seem to conform to the typical pattern I expect to see. I recommend com.mochamatcha.sandbox.

**Review by jwarmkessel — COMMENTED** [review]
@alvinyongho I left some comments for you.

File: Motcha/AppConfiguration.swift (hunk 3 above)
Comment (alvinyongho) [reply Motcha/AppConfiguration.swift:31]: Staging needs to be upper case to match the other scheme names. (Release, Debug, Staging)

File: Motcha/AppDelegate.swift (hunk 4 above)
Comment (alvinyongho) [reply Motcha/AppDelegate.swift:29]: we are not defaulting to staging environment. the scheme is set by changing the build configuration.

**Pull request closed without merging**